redfish-core/lib/cable.hpp

```diff
@@ -0,0 +1,147 @@
+{
+using ManagedPropertyType =
+    boost::container::flat_map<std::string, std::variant<std::string, double>>;
+
+/**
+ * @brief Get Cable properties
+ * @param[in,out]   asyncResp   Async HTTP response.
+ * @param[in]       resp        A map to hold Cable properties
+ * @param[in]       cableId     Cable Id
+ */
+inline void
+    getCableProperties(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                       ManagedPropertyType& resp, const std::string& cableId)
+{
+    for (auto& propertyMap : resp)
+    {
+        if (propertyMap.first == "CableTypeDescription")
+        {
+            std::string* cableTypeDescription =
+                std::get_if<std::string>(&propertyMap.second);
+            if (cableTypeDescription == nullptr)
+            {
+                messages::internalError(asyncResp->res);
+                return;
+            }
+            asyncResp->res.jsonValue["CableType"] = *cableTypeDescription;
+        }
+        else if (propertyMap.first == "Length")
+        {
+            double* cableLength = std::get_if<double>(&propertyMap.second);
+            if (cableLength == nullptr)
+            {
+                messages::internalError(asyncResp->res);
+                return;
+            }
+            asyncResp->res.jsonValue["LengthMeters"] = *cableLength;
+        }
+    }
+
+    asyncResp->res.jsonValue["@odata.type"] = "#Cable.v1_0_0.Cable";
+    asyncResp->res.jsonValue["@odata.id"] = "/redfish/v1/Cables/" + cableId;
+    asyncResp->res.jsonValue["Id"] = cableId;
+    asyncResp->res.jsonValue["Name"] = "Cable Id";
+}
+
+/**
+ * The Cable schema
+ */
+inline void requestRoutesCable(App& app)
+{
+    BMCWEB_ROUTE(app, "/redfish/v1/Cables/<str>/")
+        .privileges(redfish::privileges::getCable)
+        .methods(boost::beast::http::verb::get)(
+            [](const crow::Request&,
+               const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+               const std::string& cableId) {
+                BMCWEB_LOG_DEBUG << "Cable Id =" << cableId;
+                auto respHandler = [asyncResp,
+                                    cableId](const boost::system::error_code ec,
+                                             ManagedPropertyType& resp) {
+                    if (ec.value() == EBADR)
+                    {
+                        messages::resourceNotFound(asyncResp->res, "Cables",
+                                                   cableId);
+                        return;
+                    }
+
+                    if (ec)
+                    {
+                        BMCWEB_LOG_DEBUG << "DBUS response error " << ec;
+                        messages::internalError(asyncResp->res);
+                        return;
+                    }
+
+                    getCableProperties(asyncResp, resp, cableId);
+                };
+
+                crow::connections::systemBus->async_method_call(
+                    respHandler, "xyz.openbmc_project.Inventory.Manager",
```

Comment:
Upstream we will not be able to call Inventory Manager and will instead call the mapper. 
For downstream, I guess I am fine here.